R/rvar-rfun.R

```diff
@@ -0,0 +1,250 @@
+#' Create functions of random variables
+#'
+#' Function that create functions that can accept and/or produce [`rvar`]s.
+#'
+#' @param .f A function (or a one-sided formula representing a function that can be parsed by
+#' [rlang::as_function()]) to turn into a function that accepts and/or produces random variables.
+#' @param rvar_args The arguments of `.f` that should be allowed to accept [`rvar`]s as arguments.
+#' If `NULL` (the default), all arguments to `.f` are turned into arguments that accept
+#' [`rvar`]s.
+#' @param ndraws When no [`rvar`]s are supplied as arguments to the returned function, this is the number
+#' of draws that will be used to construct new random variables. If `NULL`,
+#' `getOption("posterior.rvar_ndraws")` is used (default 4000). If any arguments to the returned function contain
```

Comment:
Yeah, I've used that style in the past and have come to dislike it slightly. I find it cumbersome to write APIs against, because if you wrap a function `f` with that has a parameter `x` defined by an expression with a function `g` that has a different default for `x`, if your user wants to make a call to `g` that uses the default of `x` from `f` they have to recreate the entire expression instead of passing a simpler value. 

But this doesn't bother me that much in this case (that usage would be rare here). I'll switch to the other style if you prefer it.

Reply:
I see. Both options are fine by me. Feel free to leave it as is.